data/users.js

```diff
@@ -1,8 +1,65 @@
+const { ObjectID } = require('bson');
 const mongoCollections = require('../config/mongoCollections');
+const users = mongoCollections.users;
+const bcrypt = require('bcrypt');
 const getUsersCollection = mongoCollections.users;
-const validate = require('../validations/data');
 const validations = require("../validations/validations");
 const {ObjectId} = require("mongodb");
+const validate = require('../validations/data');
+
+async function getPersonById(id){
+    // In this function we need to find the user for a given id and return them.
+    const userCollection = await users();
+    console.log(id);
+    const person = await userCollection.findOne({ _id: ObjectID(id) });
+    // Return the whole person
+    return person;
+}
+
+async function updateUser(id, name, bio, age, searchTags, mentorRelations, menteeRelations, myPosts){
+    checks(name, bio, age, searchTags, mentorRelations, menteeRelations, myPosts);
+    // In this function we need to update the information for a user corresponding to the given id.
+    let person = getPersonById(id);
+    let username = person.username;
+    let password = person.password;
+    // I need to get the password/username for the given id to put in updateduser here
```

Comment:
For the update, we are using operator - "$set".
For this operator, whatever is provided is replaced. 
You can skip providing the attributes that don't need to be updated. By which, you can skip retrieving the User for username and password.